### PR TITLE
Add simulator 

### PR DESCRIPTION
**Pull request by dlanzieri**
This PR aims to add a physical model to generate a lensing convergence map and some related tests.

### CODE REVIEW

File: sbi_lens/simulator/LogNormal_field.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  zero_freq_val: float
+   The zero point to shift the vector
+
+  Returns
+  -------  
+  power_map: Jax.DeviceArray (N,N)
+  Image
+  
+  """
+    k = 2 * jnp.pi * jnp.fft.fftfreq(N, d=field_size / N)
```
Comment (Justinezgh) [inline sbi_lens/simulator/LogNormal_field.py:71]: field_size is not defined

Comment (Justinezgh) [reply sbi_lens/simulator/LogNormal_field.py:71]: I mean you called it map_size I guess but you didn't change it in the function

Comment (dlanzieri) [reply sbi_lens/simulator/LogNormal_field.py]: Yep, fixed!

File: sbi_lens/simulator/LogNormal_field.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  power_map: Jax.DeviceArray (N,N)
+  Image
+  
+  """
+    k = 2 * jnp.pi * jnp.fft.fftfreq(N, d=map_size / N)
+    kcoords = jnp.meshgrid(k, k)
+    k = jnp.sqrt(kcoords[0]**2 + kcoords[1]**2)
+    ps_map = pk_fn(k)
+    ps_map = ps_map.at[0, 0].set(zero_freq_val)
+    power_map = ps_map * (N / field_size)**2
```
Comment (Justinezgh) [inline sbi_lens/simulator/LogNormal_field.py:76]: and here too ^^

Comment (dlanzieri) [reply sbi_lens/simulator/LogNormal_field.py]: it should be fine now!

**Comment by Justinezgh** [comment]
Ok now I'm implementing the ```get_true_posterior()``` function and so I tried to run an mcmc from this model so with omega_c and sigma_8 defined as follows: 

``` python 
theta = numpyro.sample(
        'theta',
        dist.Independent(
            dist.Normal(jnp.array([0.3, 0.8]), 0.05 * jnp.ones(2)), 1))
``` 

and got these results 
image
which are not good. 

In his model, Francois defined omega_c and sigma_8 in this way: 

``` python 
  omega_c = numpyro.sample('omega_c', dist.Normal(0., 1.0)) * 0.05 + 0.3
  sigma_8 = numpyro.sample('sigma_8', dist.Normal(0., 1.0)) * 0.05 + 0.8
```

and with that the mcmc gives us the correct posterior: 
image

( @EiffL can you remind me again why the reparametrization trick allows to obtain the correct posterior ?)


SO either we change a bit the model or I have to redefine a model just for my function but it's not very clean :/ , what do you think ?

**Comment by EiffL** [comment]
Yeah ok, so, there is a cleaner way to do this reparam thing with numpyro. I was doing something very hacky before.


The reason why this trick helps is that it puts all variables in the model on the same scale with respect to their prior. Remember that there are other variables beyond the cosmo params, which are the latent variables of the field. These are drawn from a standard unit gaussian.
If you don't rescale, the algorithm will need very small step size to handle the cosmo params, and then the chain will be very slow and not mix well because it will take many steps to change the latent variables significantly.

**Comment by EiffL** [comment]
I think it's this handler https://num.pyro.ai/en/stable/handlers.html#reparam

**Comment by EiffL** [comment]
I use it in this notebook for instance https://github.com/DifferentiableUniverseInitiative/jax-cosmo-paper/blob/master/notebooks/VectorizedNumPyro.ipynb

**Pull request merged**